src/unix/aix-common.c

```diff
@@ -253,15 +278,34 @@ int uv_interface_addresses(uv_interface_address_t** addresses, int* count) {
     sa_addr = (struct sockaddr_dl*) &p->ifr_addr;
     memcpy(address->phys_addr, LLADDR(sa_addr), sizeof(address->phys_addr));
 
-    if (ioctl(sockfd, SIOCGIFNETMASK, p) == -1) {
-      uv__close(sockfd);
-      return UV_ENOSYS;
-    }
-
-    if (inet6)
-      address->netmask.netmask6 = *((struct sockaddr_in6*) &p->ifr_addr);
-    else
+    if (inet6) {
+      memset((char*) &if6, 0, sizeof(if6));
+      strncpy(if6.ifr_name, p->ifr_name, sizeof(if6.ifr_name) - 1);
```

Comment:
`uv__strscpy()`? Or do the remaining bytes need to be zeroed out? Can you add an explaining comment in that case?

Reply:
Thanks. I thought there was a uv strcpy routine but I couldn't remember what it was called. Updated.